### PR TITLE
build(docker): map database port and add JAVA_OPTS for extra configs

### PR DESCRIPTION
**Pull request by annibalsilva**
# Description

Map the database port and add a new environment variable JAVA_OPTS (to be used to pass extra configuration when running the application, such as allowing debugging) to the compose file.

## Type of change

- [ ] Local development environment

# How Has This Been Tested?

- [ ] Run the application using the compose file in the root of the project with `JAVA_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005` and try to debug the application from the IDE
- [ ] Access the database from the host using the 5432 port

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!
Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-backend-55-backend.apps.silver.devops.gov.bc.ca/) available
Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-backend/actions/workflows/merge-main.yml)

---
Thanks for the PR!
Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-backend-55-backend.apps.silver.devops.gov.bc.ca/) available
Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-backend/actions/workflows/merge-main.yml)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ services:
       POSTGRES_PASSWORD: ${POSTGRESQL_PASSWORD}
       POSTGRES_DB: ${POSTGRESQL_DATABASE}
     hostname: database
+    ports:
+      - "5432:5432"
```
[the remark this answers is not in the document]
Comment (annibalsilva) [reply docker-compose.yml:13]: It's just so we can easily access the database from the host machine (using DBeaver, for instance) during local development.

[the remark this answers is not in the document]
Comment (RMCampos) [reply docker-compose.yml:13]: Oh ok. But this file is intended to be used together with the backend, isn't? One option would be create a `docker-compose` file only with database.

[the remark this answers is not in the document]
Comment (annibalsilva) [reply docker-compose.yml:13]: There's one with just the database already, under `database/`, which maps the port like this. I thought the difference between the database configuration in both files was that the one in the root of the project was meant to be used in some part of the deployment, but Derek said it wasn't (and wouldn't be), and that the compose files are there to be used for local development. And if they're meant for local development, we might as well map the port here too.

[the remark this answers is not in the document]
Comment (RMCampos) [reply docker-compose.yml:13]: Oh got it. Thanks for the explanation. That makes sense.

**Review by RMCampos — APPROVED** [review]
Good job 🎉

**Pull request merged**